lib/bolt/executor.rb

```diff
@@ -37,18 +37,18 @@ def on(nodes, callback = nil)
         pool.post do
           result =
             begin
+              @notifier.notify(callback, node, type: :node_start) if callback
               node.connect
               yield node
-            rescue Bolt::Node::BaseError => ex
-              Bolt::ErrorResult.new(ex.message, ex.issue_code, ex.kind)
             rescue StandardError => ex
-              node.logger.error(ex)
-              Bolt::ExceptionResult.new(ex)
+              Bolt::Result.from_exception(ex)
             ensure
               node.disconnect
             end
           results[node] = result
-          @notifier.notify(callback, node, result) if callback
+          if callback
```

Comment:
Why do we make this optional? I see `callback = nil` in args, but if we never exercised it I'm not sure why we want to preserve that.

Reply:
That is how the boltlib functions call it
https://github.com/puppetlabs/bolt/blob/master/modules/boltlib/lib/puppet/functions/run_task.rb#L76